doc/design.md

```diff
@@ -273,6 +293,121 @@ Returns:
     * tbd
 
 
+### **`POST`** `/oeci_login` ?
+
+## Question
+
+What's the workflow for accessing OECI?
+
+Is it a webpage redirect, like the frontend prototype?
+
+Or is it an authentication serverside, like the current Crawler implementation?
+These behaviors need to be reconciled.
+
+I see three design versions:
+
+- Rely on a frontend page redirect, which obtains login access from the OECI directly to allow subsequent requests by Crawler. This requires a Crawler redesign, which currently passes creds to the remote site. It also requires caching a login token provided by OECI. Does the OECI login allow that, e.g. use a cookie?
```

Comment:
We will not be holding onto the cookie (at least in the backend) sent back from OECI for multiple requests. We can create our own cookie to set on the frontend that can be used for each request. I'm not sure but maybe we could send the OECI cookie to the client to send back on further requests, but I prefer using our own cookie so that we have more control and less dependencies. FYI: the OECI cookie expires after ~5mins of inactivity.

Reply:
From what I can tell on the Crawler side, it looks like the flow is to instantiate the Crawler, call crawler.login(oeci_username, oeci_password), and then can make subsequent search calls. when the crawler gets reinitialized (i.e. with every /search/ from the frontend), it needs to run crawler.login() again. 

That means we never directly / explicitly do anything with an OECI cookie.

@NickSchimek does this sound right?

So the details to be worked out on exactly how the oeci (encrypted) credentials get passed back and forth with the oeci_login response and with the  /search/ request/response, but it does look like the login-then-search crawler operations need to be run on every /search/

Pinging @ash-tamraz because he's working on /search/, I'm wondering what you think?

Reply:
@ForrestLonganecker and I are looking into best practices with this workflow , and low-level implementation of cookie usage both on the frontend and backend.

Reply:
Ok, to correct myself from earlier:  looking more closely at the crawler, I see that the oeci cookie is set to the crawler's Session object by a successful oeci login request. It's then sent automatically with subsequent requests. That's all just handled by python's Requests package, #TIL.